app/lib/staypuft/network_query.rb

```diff
@@ -155,6 +155,7 @@ def interface_hash_for_host(subnet_type_name, host)
     def gateway_hash_for_host(host)
       gateway_hash = interface_hash_for_host(Staypuft::SubnetType::PUBLIC_API, host)
       gateway_hash = interface_hash_for_host(Staypuft::SubnetType::PXE, host) unless gateway_hash[:subnet]
+      gateway_hash = interface_hash_for_host(Staypuft::SubnetType::EXTERNAL, host) if Staypuft::Role.compute.any? { |r| r.name == host.hostgroup.name } && host.deployment.nova_networking?
```

Comment:
Hmm. this looks like this will set the gateway to EXTERNAL if there are any compute roles defined. Since roles are defined statically via seed data, this would *always* be true. I'd think you would want to make sure that this host belongs to a compute role. You also probably want to put the PXE rule *last* -- that way if external isn't defined, you at least get an internal gatway (vs. none) -- that's why that PXE line was last with the "unless gateway_hash..." bit was there.